### PR TITLE
feat: Improve-terraform-plan with AKS connection

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### List of changes

- added aks kubeconfig generation to allow terraform plan
- added check parameters
- improve readme

### Motivation and context

Allow terraform to make plan for kubernetes projects

### Type of changes

- [ ] Add new template
- [x] Update template configuration
- [ ] Remove template
- [ ] Other changes

### Other information

---

### CODE REVIEW

**Review by pasqualedevita — COMMENTED** [review]
after tf plan/apply add a
```
rm -rf .kube
rm -rf .azure
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: templates/terraform-plan/template.yaml [hunk 1]
```diff
@@ -1,35 +1,129 @@
...
-    displayName: 'Azure service connection name.'
+  - name: 'AZURE_SERVICE_CONNECTION_NAME'
+    displayName: '(Required) Azure service connection name.'
     type: string
     default: "none"
   - name: 'ENVIRONMENT'
-    displayName: 'Environment to apply to. Possible values are [dev|uat|prod] etc.'
+    displayName: '(Required) Environment to apply to. Possible values are [dev|uat|prod] etc.'
     type: string
     default: "dev"
```
Comment (pasqualedevita) [inline templates/terraform-plan/template.yaml:10]:
```suggestion
```
it's required so we can remove the default value

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: templates/terraform-plan/template.yaml (hunk 1 above)
Comment (diegolagospagopa) [reply templates/terraform-plan/template.yaml:10]: ok

File: templates/terraform-plan/README.md [hunk 2]
```diff
@@ -1,28 +1,71 @@
...
+          # Run terraform plan idpay-app
+          - template: templates/terraform-plan/template.yaml@terraform
+            parameters:
+              ENVIRONMENT: "dev"
+              WORKINGDIR: 'src/domains/idpay-app'
+              AZURE_SERVICE_CONNECTION_NAME: DEV-CSTAR-SERVICE-CONN
+              AKS_NAME: 'cstar-d-weu-dev01-aks'
+              AKS_API_SERVER_URL: ${{ variables.dev01AksApiServerUrl }}
+              AKS_AZURE_DEVOPS_SA_CA_CRT: ${{ variables.dev01AksAzureDevopsSACacrt }}
+              AKS_AZURE_DEVOPS_SA_TOKEN: ${{ variables.dev01AksAzureDevopsSAToken }}
```
[the remark this answers is not in the document]
Comment (diegolagospagopa) [reply templates/terraform-plan/README.md:70]: Ok fixed

**Review by pasqualedevita — DISMISSED** [review]

**Pull request merged**